packages/docusaurus-logger/src/index.ts

```diff
@@ -0,0 +1,134 @@
+
+const path: import('picocolors/types').Formatter = (msg) =>
+  pico.cyan(pico.underline(msg));
+const id: import('picocolors/types').Formatter = (msg) =>
+  pico.blue(pico.bold(msg));
+const code: import('picocolors/types').Formatter = (msg) =>
+  pico.cyan(`\`${msg}\``);
+const subdue = pico.gray;
+const num = pico.yellow;
+
+function interpolate(
+  msgs: TemplateStringsArray,
+  ...values: InterpolatableValue[]
+): string {
+  let res = '';
+  values.forEach((value, idx) => {
+    const flag = msgs[idx].match(/%[a-z]+$/);
+    res += msgs[idx].replace(/%[a-z]+$/, '');
+    const format = (function () {
+      if (!flag) {
+        return (a: string | number) => a;
+      }
+      switch (flag[0]) {
+        case '%p':
+          return path;
+        case '%n':
+          return num;
+        case '%i':
+          return id;
+        case '%s':
+          return subdue;
+        case '%c':
+          return code;
+        default:
+          throw new Error(
+            'Bad Docusaurus logging message. This is likely an internal bug, please report it.',
+          );
+      }
+    })();
+    res += Array.isArray(value)
+      ? `\n- ${value.map((v) => format(v)).join('\n- ')}`
+      : format(value);
+  });
+  res += msgs.slice(-1)[0];
+  return res;
+}
+
+function info(msg: any): void;
+function info(
+  msg: TemplateStringsArray,
+  ...values: [InterpolatableValue, ...InterpolatableValue[]]
+): void;
+function info(msg: any, ...values: InterpolatableValue[]): void {
+  console.info(
+    `${pico.cyan(pico.bold('[INFO]'))} ${
+      values.length === 0
+        ? msg
+        : interpolate(msg as TemplateStringsArray, ...values)
+    }`,
+  );
+}
+function warn(msg: any): void;
+function warn(
+  msg: TemplateStringsArray,
+  ...values: [InterpolatableValue, ...InterpolatableValue[]]
+): void;
+function warn(msg: any, ...values: InterpolatableValue[]): void {
+  console.warn(
+    pico.yellow(
+      `${pico.bold('[WARNING]')} ${
+        values.length === 0
+          ? msg
+          : interpolate(msg as TemplateStringsArray, ...values)
+      }`,
+    ),
+  );
+}
+function error(msg: any): void;
+function error(
+  msg: TemplateStringsArray,
+  ...values: [InterpolatableValue, ...InterpolatableValue[]]
+): void;
+function error(msg: any, ...values: InterpolatableValue[]): void {
+  console.error(
+    pico.red(
+      `${pico.bold('[ERROR]')} ${
+        values.length === 0
+          ? msg
+          : interpolate(msg as TemplateStringsArray, ...values)
+      }`,
+    ),
+  );
+}
+function success(msg: any): void;
+function success(
+  msg: TemplateStringsArray,
+  ...values: [InterpolatableValue, ...InterpolatableValue[]]
+): void;
+function success(msg: any, ...values: InterpolatableValue[]): void {
+  console.log(
+    `${pico.green(pico.bold('[SUCCESS]'))} ${
+      values.length === 0
+        ? msg
+        : interpolate(msg as TemplateStringsArray, ...values)
+    }`,
+  );
+}
+
+const logger = {
+  ...pico,
```

Comment:
the point of this abstraction is actually to not expose the underlying lib to our own code, so pico/chakl should not be exposed

Reply:
We occasionally need to use the imperative APIs (`red`, `bold`) as opposed to the semantic declarative ones (`path`, `id`). So the alternative to re-exposing `pico` is exporting every color, I'm fine with that as well

Reply:
> So the alternative to re-exposing pico is exporting every color, I'm fine with that as well

We should only expose colors that we actually use, it's like a "terminal design system / design tokens".

We really don't want to expose pico/chalk/whatever in any way here